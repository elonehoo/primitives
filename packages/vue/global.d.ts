import type { AxeMatchers } from 'vitest-axe/matchers'
/// <reference types="vite/client" />
import 'vitest'

declare module 'vitest' {
  export interface AsymmetricMatchersContaining extends AxeMatchers {}
  export interface Assertion extends AxeMatchers {}
}
