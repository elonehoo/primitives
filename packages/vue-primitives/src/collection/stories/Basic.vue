<script setup lang="ts">
import List from './List.vue'
import LogItems from './LogItems.vue'
import Item from './item.vue'
</script>

<template>
  <List>
    <Item>Red</Item>
    <Item disabled>
      Green
    </Item>
    <Item>Blue</Item>
    <LogItems />
  </List>
</template>
