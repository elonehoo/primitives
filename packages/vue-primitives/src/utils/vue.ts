import type { Ref } from 'vue'

export function forwardRef<T = HTMLElement>(elRef: Ref<T>) {
  function setRef(nodeRef: any) {
    elRef.value = nodeRef?.$el
  }

  return setRef
}

export function composeEventHandlers<E extends Event>(
  originalEventHandler?: (event: E) => void,
  ourEventHandler?: (event: E) => void,
  { checkForDefaultPrevented = true } = {},
) {
  return function handleEvent(event: E) {
    originalEventHandler?.(event)

    if (checkForDefaultPrevented === false || !((event as unknown) as Event).defaultPrevented)
      ourEventHandler?.(event)
  }
}
