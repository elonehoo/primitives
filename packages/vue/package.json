{
  "name": "vue-packages",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@rollup/plugin-alias": "^5.1.1"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.1.4",
    "@vueuse/core": "^10.11.1",
    "filter-anything": "^3.0.7",
    "typescript": "^5.6.3",
    "unplugin-vue-components": "^0.26.0",
    "vite-plugin-pages": "^0.32.3",
    "vue-router": "^4.4.5",
    "vue-tsc": "^2.1.6"
  }
}
