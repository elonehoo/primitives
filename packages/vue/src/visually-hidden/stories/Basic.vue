<script setup lang="ts">
import { OkuVisuallyHidden } from '@oku-ui/visually-hidden'
</script>

<template>
  <button>
    <OkuVisuallyHidden>Save the file</OkuVisuallyHidden>
    <span :aria-hidden="true">ðŸ’¾</span>
  </button>
</template>
