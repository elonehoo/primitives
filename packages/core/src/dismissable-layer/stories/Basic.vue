<script setup lang="ts">
import { shallowRef } from 'vue'
import { DismissableLayer } from '../index.ts'

const open = shallowRef(false)
const openButtonRef = shallowRef<HTMLElement>()

const dismissOnEscape = shallowRef(false)
const dismissOnPointerDownOutside = shallowRef(false)
const dismissOnFocusOutside = shallowRef(false)
const disabledOutsidePointerEvents = shallowRef(false)

function openAlert() {
  // eslint-disable-next-line no-alert
  window.alert('hey!')
}
</script>

<template>
  <div>
    <div :style="{ fontFamily: 'sans-serif', textAlign: 'center' }">
      <h1>DismissableLayer</h1>

      <div :style="{ display: 'inline-block', textAlign: 'left', marginBottom: 20 }">
        <label :style="{ display: 'block' }">
          <input
            type="checkbox"
            :checked="dismissOnEscape"
            @change="(event: any) => dismissOnEscape = event.target.checked"
          >
          {{ ' ' }}
          Dismiss on escape?
        </label>

        <label :style="{ display: 'block' }">
          <input
            type="checkbox"
            :checked="dismissOnPointerDownOutside"
            @change="(event: any) => dismissOnPointerDownOutside = (event.target.checked)"
          >{{ ' ' }}
          Dismiss on pointer down outside?
        </label>

        <label :style="{ display: 'block' }">
          <input
            type="checkbox"
            :checked="dismissOnFocusOutside"
            @change="(event: any) => dismissOnFocusOutside = (event.target.checked)"
          >{{ ' ' }}
          Dismiss on focus outside?
        </label>

        <hr>

        <label :style="{ display: 'block' }">
          <input
            type="checkbox"
            :checked="disabledOutsidePointerEvents"
            @change="(event: any) => disabledOutsidePointerEvents = (event.target.checked)"
          >{{ ' ' }}
          Disable outside pointer events?
        </label>
      </div>

      <div :style="{ marginBottom: 20 }">
        <button ref="openButtonRef" type="button" @click="() => open = !open">
          {{ open ? 'Close' : 'Open' }} layer
        </button>
      </div>

      <DismissableLayer
        v-if="open"
        :disable-outside-pointer-events="disabledOutsidePointerEvents"
        :style="{
          display: 'inline-flex',
          justifyContent: 'center',
          alignItems: 'center',
          verticalAlign: 'middle',
          width: '400px',
          height: '300px',
          backgroundColor: 'black',
          borderRadius: '10px',
          marginBottom: '20px',
        }"
        @dismiss="() => open = false"
        @escape-keydown="(event) => {
          if (dismissOnEscape === false) {
            event.preventDefault();
          }
        }"
        @pointerdown-outside="(event) => {
          if (dismissOnPointerDownOutside === false || event.target === openButtonRef) {
            event.preventDefault();
          }
        }"
        @focus-outside="(event) => {
          if (dismissOnFocusOutside === false) {
            event.preventDefault();
          }
        }"
      >
        <input type="text">
      </DismissableLayer>

      <div :style="{ marginBottom: '20px' }">
        <input type="text" value="hello" :style="{ marginRight: '20px' }">
        <button
          type="button"
          @mousedown="openAlert"
        >
          hey!
        </button>
      </div>
    </div>
  </div>
</template>
