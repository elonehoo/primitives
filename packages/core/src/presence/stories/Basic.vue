<script setup lang="ts">
import { shallowRef } from 'vue'
import { usePresence } from '../index.ts'
import './styles.css'

const open = shallowRef(true)

function setOpen() {
  open.value = !open.value
}

const elRef = shallowRef<HTMLElement>()

const isPresent = usePresence(elRef, open)
</script>

<template>
  <button @click="setOpen">
    toggle
  </button>

  <div v-if="isPresent" ref="elRef">
    Content
  </div>
</template>
