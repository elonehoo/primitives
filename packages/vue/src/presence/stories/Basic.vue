<script setup lang="ts">
import { OkuPresence } from '@oku-ui/presence'
import { ref } from 'vue'

const open = ref(true)

function toggle() {
  open.value = !open.value
}
</script>

<template>
  <button @click="toggle">
    toggle
  </button>
  <OkuPresence :present="open">
    <div>
      Content
    </div>
  </OkuPresence>
</template>
