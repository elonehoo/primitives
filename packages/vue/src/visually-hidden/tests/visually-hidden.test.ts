import { h } from 'vue'
import { enableAutoUnmount, mount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuVisuallyHidden } from '../'

import Basic from '../stories/Basic.vue'

enableAutoUnmount(afterEach)

describe('okuVisuallyHidden', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = mount(OkuVisuallyHidden)
  })

  it('should render OkuVisuallyHidden correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  it('should render as a span element', () => {
    wrapper = mount(OkuVisuallyHidden, {
      attachTo: document.body,
    })

    expect(wrapper.element.tagName).toBe('SPAN')
  })

  it('should have the default style', () => {
    wrapper = mount({
      setup() {
        return () => h(OkuVisuallyHidden)
      },
    })

    expect(wrapper.attributes('style')).toBe(
      'position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;',
    )
  })
})

describe('okuVisuallyHidden Stories', () => {
  describe('basic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Basic>>
    const TEXT_CHILD = 'Save the file'

    beforeEach(async () => {
      wrapper = mount(Basic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render OkuVisuallyHidden correctly', () => {
      expect(wrapper.findComponent(OkuVisuallyHidden).exists()).toBe(true)
    })

    it('should render OkuVisuallyHidden as a span element', () => {
      expect(wrapper.findComponent(OkuVisuallyHidden).element.tagName).toBe('SPAN')
    })

    it('should have default style for OkuVisuallyHidden', () => {
      expect(wrapper.findComponent(OkuVisuallyHidden).attributes('style')).toBe(
        'position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;',
      )
    })

    it('should render text inside OkuVisuallyHidden', () => {
      expect(wrapper.findComponent(OkuVisuallyHidden).text()).toBe(TEXT_CHILD)
    })

    it('should have correct attributes for the second span', () => {
      expect(wrapper.findAll('span')[1].exists()).toBe(true)
      expect(wrapper.findAll('span')[1].element.tagName).toBe('SPAN')
      expect(wrapper.findAll('span')[1].attributes('aria-hidden')).toBe('true')
    })
  })
})
