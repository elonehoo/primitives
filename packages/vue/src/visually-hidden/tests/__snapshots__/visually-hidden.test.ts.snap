// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuVisuallyHidden > should render OkuVisuallyHidden correctly 1`] = `"<span style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"></span>"`;

exports[`okuVisuallyHidden Stories > basic > should render correctly 1`] = `"<button><span style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;">Save the file</span><span aria-hidden="true">ğŸ’¾</span></button>"`;
